### ISSUE TITLE
GraphQL: block doesn't correctly return block on canonical chain. Change the API to choose the block on the canonical chain

### ISSUE THREAD

**Issue by garethtdavies** [MinaProtocol/mina#8111]
Getting a block by height doesn't return the block on the canonical chain.

```
{
  block(height: 518) {
    protocolState {
      consensusState {
        blockHeight
      }
      previousStateHash
    }
    stateHash
  }
}
```

Returns:

```
{
  "data": {
    "block": {
      "protocolState": {
        "consensusState": {
          "blockHeight": "518"
        },
        "previousStateHash": "3NKmuEHwrUALc38yusnxd8ZWjuapvzFKBpuoHVRDxgw8EBP9d9uh"
      },
      "stateHash": "3NKmAD9hc7jQcnC8KnKF8o4HSddcfTL2GiPD3PVDWSXt8USe5ciB"
    }
  }
}
```

Note, the previousStateHash is 3NKmuEHwrUALc38yusnxd8ZWjuapvzFKBpuoHVRDxgw8EBP9d9uh

Getting the block at 517:

```
{
  block(height: 517) {
    protocolState {
      consensusState {
        blockHeight
      }
      previousStateHash
    }
    stateHash
  }
}
```
This returns the block with stateHash 3NKUnZCLASJ9BGkyZFegZDLt85okDmS1oipQrC8KtmemobHEYgHX

```
{
  "data": {
    "block": {
      "protocolState": {
        "consensusState": {
          "blockHeight": "517"
        },
        "previousStateHash": "3NL2oaTUMuvVt67svBXqcxZpkfRtU8grBFhvbHDzPzebP4Xxbwfe"
      },
      "stateHash": "3NKUnZCLASJ9BGkyZFegZDLt85okDmS1oipQrC8KtmemobHEYgHX"
    }
  }
}
```

Using the `bestChain` query showing only the relevant blocks it looks like it is returning the wrong block for 517. 

```
      {
        "protocolState": {
          "consensusState": {
            "blockHeight": "517"
          },
          "previousStateHash": "3NL2oaTUMuvVt67svBXqcxZpkfRtU8grBFhvbHDzPzebP4Xxbwfe"
        },
        "stateHash": "3NKmuEHwrUALc38yusnxd8ZWjuapvzFKBpuoHVRDxgw8EBP9d9uh"
      },
      {
        "protocolState": {
          "consensusState": {
            "blockHeight": "518"
          },
          "previousStateHash": "3NKmuEHwrUALc38yusnxd8ZWjuapvzFKBpuoHVRDxgw8EBP9d9uh"
        },
        "stateHash": "3NKmAD9hc7jQcnC8KnKF8o4HSddcfTL2GiPD3PVDWSXt8USe5ciB"
      },
```

**Comment by deepthiskumar** [comment]
Yeah it it returning *a* block at the given height. Should either return the block in the best chain or all the blocks at the given height

**Comment by bkase** [comment]
Oh wow, that definitely needs fixing

**Comment by garethtdavies** [comment]
> Yeah it it returning _a_ block at the given height. Should either return the block in the best chain or all the blocks at the given height

Should be the canonical block only in my opinion

**Comment by psteckler** [comment]
Fixed by #8114.
